### PR TITLE
Add document on CPU power and performance

### PR DESCRIPTION
**Pull request by puneetse**
Adds documentation about CPU performance governor and thermald.

Closes https://github.com/clearlinux/clear-linux-documentation/issues/512

### CODE REVIEW

File: source/clear-linux/guides/maintenance/cpu-performance.rst [hunk 1]
```diff
@@ -0,0 +1,165 @@
+.. _cpu-performance:
+
+CPU Power and Performance
+#########################
+
+Modern x86 CPUs employ a numer of features and technologies to balance
```
Comment (tpleavitt) [inline source/clear-linux/guides/maintenance/cpu-performance.rst]: numer -> number

File: source/clear-linux/guides/maintenance/cpu-performance.rst [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+performance, energy, and thermal efficiencies.
+
+By default, |CL| prioritizes maximum CPU performance with the philosophy that
+the faster programs finish execution, the faster the CPU can return to an idle
```
Comment (tpleavitt) [inline source/clear-linux/guides/maintenance/cpu-performance.rst]: suggest change to "the faster a program finishes"

File: source/clear-linux/guides/maintenance/cpu-performance.rst [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+.. contents:: :local:
+    :depth: 2
+
+
+
+CPU clock frequency scaling
+===========================
+
+The CPUFreq subsystem in Linux allows the OS to control C-states and P-states
+via scaling governors and drivers that provide an algorithms defining how and
```
Comment (tpleavitt) [inline source/clear-linux/guides/maintenance/cpu-performance.rst]: suggest to remove the "an" in an algorithms

File: source/clear-linux/guides/maintenance/cpu-performance.rst [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+voltages based on CPU demand.
+
+P-states can typically be limited or disabled in a system's firmware (BIOS/EFI).
+
+Turbo boost
+~~~~~~~~~~~
+
+Turbo boost, found on some modern Intel CPUs, allows core(s) on a processor to
+temporarily operate at a higher than rated CPU clock frequency to accomodate
+demanding workloads if the CPU is under defined power and thernal thresholds.
```
Comment (tpleavitt) [inline source/clear-linux/guides/maintenance/cpu-performance.rst]: thernal -> thermal

File: source/clear-linux/guides/maintenance/cpu-performance.rst [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+~~~~~~~~~~~
+
+Turbo boost, found on some modern Intel CPUs, allows core(s) on a processor to
+temporarily operate at a higher than rated CPU clock frequency to accomodate
+demanding workloads if the CPU is under defined power and thernal thresholds. 
+
+Turbo boost is an extension of P-states. As such, changing or limiting C-states
+or P-states impact the ability of a process to enter Turbo boost. 
+
+Turbo boost can be disabled in a system's EFI or BIOS. Turbo boost can also be
```
Comment (tpleavitt) [inline source/clear-linux/guides/maintenance/cpu-performance.rst]: just a note...EFI normally refers to the Intel-specific EFI 1.x BIOS architecture, UEFI refers to the unified 2.x+ version that has been adopted by virtually everyone.  Recommend using UEFI when talking about the UEFI firmware

File: source/clear-linux/guides/maintenance/cpu-performance.rst [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+.. contents:: :local:
+    :depth: 2
+
+
+CPU clock frequency scaling
+===========================
+
+The CPUFreq subsystem in Linux allows the OS to control C-states and P-states
+via CPU drivers and governors that provide algorithms defining how and when to
+enter these states.
```
Comment (ahkok) [inline source/clear-linux/guides/maintenance/cpu-performance.rst]: Lacking any explanation of what C-states are. Which would really help to explain some of the notes.

Comment (ahkok) [reply source/clear-linux/guides/maintenance/cpu-performance.rst]: I think you need to explain the concepts at the top, then dive into the technical details. Now you're mixing it around.

File: source/clear-linux/guides/maintenance/cpu-performance.rst [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+  The intel_pstate driver only supports *performance* and *powersave* governors.
+
+
+Thermal management
+==================
+
+`thermald`_ is a Linux thermal management daemon used to prevent the
+overheating of platforms. It monitors temperature and controls available
+cooling methods accordingly. This can be especially desirable for laptops.
```
Comment (ahkok) [inline source/clear-linux/guides/maintenance/cpu-performance.rst]: doesn't explain how it does this - specifically, it inserts sleep cycles (forces the cpu into a C-state).

File: source/clear-linux/guides/maintenance/cpu-performance.rst [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+
+By default, thermald is disabled in |CL|. thermald can be enabled using the
+systemd service by running the command: 
+
+.. code:: bash
+   
+   sudo systemctl enable thermald
+
+
+For more information, see the thermald man page :command:`man thermald`.
```
Comment (ahkok) [inline source/clear-linux/guides/maintenance/cpu-performance.rst]: Leave a note about thermalmonitor?

**Comment by ahkok** [comment]
All in all it's pretty good. I would change the organization though. First, explain the concepts of p-states, c-states, turbo and include the needed background. Then list all the driver/sysfs/kernel details one by one that are applicable.

But, it's pretty comprehensive already.

**Comment by puneetse** [comment]
@ahkok thanks. I pushed an update based on your feedback. Do you mind doing a second pass?

File: source/clear-linux/guides/maintenance/cpu-performance.rst [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+
+  The intel_pstate driver only supports *performance* and *powersave* governors.
+
+
+Thermal management
+==================
+
+`thermald`_ is a Linux thermal management daemon used to prevent the
+overheating of platforms. When temperate thresholds are exceeded, thermald
+forced a C-state and inserts CPU sleep cycles and adjusts available cooling
```
Comment (ahkok) [inline source/clear-linux/guides/maintenance/cpu-performance.rst]: forces*

and ... and .... without commas.

File: source/clear-linux/guides/maintenance/cpu-performance.rst [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+methods. This can be especially desirable for laptops.
+
+By default, thermald is disabled in |CL| and automatically is enabled if
+battery power is detected. thermald can be manually enabled using the systemd
+service by running the command: 
+
+.. code:: bash
+   
+   sudo systemctl enable thermald
```
Comment (ahkok) [inline source/clear-linux/guides/maintenance/cpu-performance.rst]: what about `start` ?

**Comment by puneetse** [comment]
@intelkevinputnam this is ready for final review. Could you take a look and merge if there is nothing blocking?

**Comment by krisdale** [comment]
@puneetse Confirming - this current version is approved by @ahkok too? I'll provide final review :)

**Comment by puneetse** [comment]
@krisdale yes.

**Pull request merged**